### DISCUSSION TITLE
Update Documentation for SSH Connection Issues During Installation

### DISCUSSION THREAD

**Discussion by spectaA** [coollabsio/coolify#2639] [Improvement Requests]
Hi Coolify Team,

I recently encountered an issue during the installation of Coolify where it was unable to connect to localhost due to SSH connection problems. The specific error message displayed was:

`root@host.docker.internal: Permission denied (publickey,password).`

After spending several hours troubleshooting, I discovered that the problem was not with Coolify itself, but with the default SSH configuration on Ubuntu Server. 

By default, the sshd_config file contains the following rule:
`AllowGroups admin`
This rule prevents the root user from connecting unless explicitly added to the AllowGroups.

To resolve this issue, I added the root group to the AllowGroups in the sshd_config file as follows:
`AllowGroups admin root`

I believe it would be beneficial for other users if this information were included in the Coolify installation documentation. This addition could save significant troubleshooting time and help ensure a smoother installation process.

Thank you for considering this request!

**Comment by palepinkdot** [comment]
Just ran into this problem myself and was able to resolve it with the solution provided.

**Comment by rafaseron** [comment]
I had this problem now, and unfortunately the AllowGroups admin line did not exist.

I solved the problem by installing Coolify one on top of the other (I ran the 'curl' command without uninstalling the existing installation)

**Comment by magnusburton** [comment]
Running in to this problem but unfortunately this does not fix the issue. Still searching for a solution

**Comment by grazy27** [comment]
After spending good 5 hrs on trying to get what's wrong, i finally got an answer. 
There's log level 'DEBUG3' for OpenSSH (!= DEBUG), in which it finally said what's wrong: 
![image](https://github.com/user-attachments/assets/3762bcc0-c43c-4d6a-bfc0-1dc090069519)


So, to fix that, run `sudo nano /etc/ssh/sshd_config`
and append(replace) the next lines

```
PermitRootLogin yes
PubkeyAcceptedAlgorithms +ssh-ed25519
HostKeyAlgorithms +ssh-ed25519
```

**Reply by Vuurbaard** [reply]
Thank you! This seemed to have done the trick on my VPS I have rented (from Contabo). ❤️

**Reply by pencilcheck** [reply]
I use this instead
```
PermitRootLogin yes
PubkeyAuthentication yes
PubkeyAcceptedKeyTypes ssh-ed25519-cert-v01@openssh.com,ssh-ed25519
```

But turns out that for some reason after reboot the private key has changed in coolify, so I had to copy what's in the Keys & Tokens to the file authorized_keys for it to work again, not sure why coolify change the keys after reboot. (I run many installation script, does it actually change the key? that is concerning)

**Comment by Smit2553** [comment]
Had a simular issue. Neither solution above worked. 

My solution was there are 2 different `.ssh/authorized_keys` files. One in root and one under my user. The `.ssh/authorized_keys` file under root had the coolify ssh key on the same line as my personal login ssh key causing it to break. 

Adding a new line between the two ssh keys fixed the issue.

**Reply by DevDams** [reply]
I had the same problem and indeed the coolify key was on the same line as my key. This solution helped me

**Comment by juandation** [comment]
HI! Still having the same issue trying to validate the server: 

"Please validate your configuration and connection.
Check this [documentation](https://coolify.io/docs/knowledge-base/server/openssh) for further help. 
Error: No query results for model [App\Models\PrivateKey] 0". 

I ran the coolify installation twice (Not manually) and the server still is not being validated. I can connect to my VPs via ssh successfully, and I can login to the coolify admin panel. Any possible solution? I tried all from this thread but nothing is working.

**Comment by balogun14** [comment]
i had a similar issue all i had to do was uninstall and reinstall again

**Reply by SayakSarkar02** [reply]
what to uninstall and reinstall?

**Comment by Roemie** [comment]
I also ran into this while self-hosting Coolify (on Unraid in my case), and for me the root cause wasn’t sshd_config but Coolify’s own private key / server registration state.

A few things that finally solved it:

Ensure the Coolify key actually exists in the DB

`docker exec -it Coolify php artisan tinker --execute="dump(App\Models\PrivateKey::all(['id','name'])->toArray());"
`

If you see Error loading private key or no rows at all, regenerate a key pair in the UI and make sure the public key gets added (on its own line) to ~/.ssh/authorized_keys on the target host.

Sentinel must be running with a valid token

```
docker run -d \
  --name coolify-sentinel \
  -e SENTINEL_TOKEN="your_random_hex_token" \
  -e SERVER_ID="localhost" \
  -v /var/run/docker.sock:/var/run/docker.sock \
  ghcr.io/coollabsio/sentinel:latest
```


Then update the server row inside Coolify with that token if the UI doesn’t expose a field yet.

Set a proper FQDN in Instance Settings — Coolify won’t let you enable Sentinel with just an IP.
Even for local installs, using a fake domain (e.g. unraid.local) mapped via /etc/hosts or --add-host in Docker parameters worked for me.

After those three steps the “No query results for model [App\Models\PrivateKey] 0” error went away, and Sentinel synced properly. From there I could validate the server and deploy apps without further SSH issues.

**Comment by madsaylor** [comment]
Lol that's amazing. 
I did One Click install by Digital Ocean and got this exact problem. And nothing is working. None of the solutions.
Destroyed droplet started another one - still the same. 
One-Click out of box install completely broken.  
Now I need to dive into PHP, ssh logs and configs. Good times.

**Comment by Jothirmai** [comment]
That “now I need to dive into PHP, SSH logs and configs” part is exactly the pain I'm working on. 😅

I've been building Thread Nimbus, an SSH client with an AI layer that can help investigate what's actually happening on a server — logs, processes, Docker containers, ports, disk/CPU issues, etc.

It's not meant to replace SSH; it's meant to make the “what the hell is happening on this server?” part much easier.

If you're still dealing with this, I'd genuinely love for you to try it and tell me whether it would have helped in this situation:
https://nimbus.threadlane.in/
